platform/root/index.ts

```diff
@@ -44,7 +44,9 @@ export class Root {
       // This specifically console.logs because we were not able to configure
       // the logger.
       console.error('Configuring logger failed:', e.message);
-      return this.shutdown(e);
+
+      await this.shutdown(e);
+      throw e;
```

Comment:
Throwing error here won't do any harm if platform is run separately since shutdown will call `process.exit` anyway, but it's essential for our current setup.